test/hooks/useFetch.test.ts

```diff
@@ -0,0 +1,47 @@
+import { renderHook, act } from '@testing-library/react-hooks';
+import { generateMockFetch } from 'test/testUtils';
+import { useFetch } from 'hooks';
+import artists from 'data/artists.json';
+import events from 'data/events.json';
+import venues from 'data/venues.json';
+import type { HakkasanResponse } from 'lib/hakkasan';
+
+describe('useFetch(endpoint)', () => {
+  test('should fetch data from the Next.js proxy API', async () => {
+    const endpoint = '/api/hakkasan?ref=events';
+    const expected = {
+      artists,
+      events,
+      venues,
+    };
+
+    global.fetch = generateMockFetch({
+      json: () => Promise.resolve(expected),
+    });
+
+    const { result } = renderHook(() => useFetch<HakkasanResponse>(endpoint));
+
+    await act(async () => {
+      await result.current.fetchData();
+    });
+
+    expect(result.current.data).toEqual(expected);
+  });
+
+  test('should set errors', async () => {
+    const endpoint = '/api/hakkasan?ref=events';
+    const expectedError = 'response.json is not a function';
+
+    global.fetch = generateMockFetch({
+      json: undefined,
+    });
```

Comment:
It would be better to mock the actual `json` method to reject with our own test error so that we don't rely on some message text from an error we don't control. For instance, if we were to update the Node version and they change the error message for when some non-function tries to be invoked, this test would fail when it shouldn't.

```suggestion
    const expectedError = 'response.json is not a function';

    global.fetch = generateMockFetch({
      json: () => Promise.reject(new Error(expectedError)),
    });
```

Reply:
Good point! Thank you! :D